client/document/support-user.jsx

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+
+function supportUserFn( { user, token, authorized } ) {
+	const url = window.location.toString();
+
+	if ( url.indexOf( '?' ) > 0 ) {
+		const cleanUrl = url.substring( 0, url.indexOf( '?' ) );
+		window.history.replaceState( {}, document.title, cleanUrl );
+	}
+
+	if ( authorized ) {
+		window.sessionStorage.setItem( 'boot_support_user', window.JSON.stringify( { user, token } ) );
+	}
+
+	window.location.replace( '/' );
+}
+
+function SupportUser( { supportUser, supportToken, authorized = false } ) {
+	return (
+		<html lang="en">
+			<body>
+				{ /* eslint-disable react/no-danger */ }
+				<script
+					dangerouslySetInnerHTML={ {
```

Comment:
I think the block below needs to be properly escaped as the parameters of this function come from untrusted user input.

Reply:
aren't the security aspects the same as they previously were?  The [old code](https://github.com/Automattic/wp-calypso/pull/22296/files#diff-72bfc6b9430b492b80541033bc3df0daL18) also received `user` and `token` from `req.query` and then output it directly into JS.

Reply:
Ah! I take it back! I think pug has been escaping it for us by default. [source](https://pugjs.org/language/interpolation.html#string-interpolation-escaped)

Reply:
@xknown is 1510562 sufficient?